api/segments/serializers.py

```diff
@@ -120,6 +126,22 @@ def _update_or_create_segment_rules(
                 child_rules, rule=child_rule, is_create=is_create
             )
 
+    def _update_or_create_metadata(self, metadata_data, segment=None):
+        for metadata_item in metadata_data:
+            metadata_id = metadata_item.pop("id", None)
+            if metadata_item.get("delete"):
+                Metadata.objects.filter(id=metadata_id).delete()
+                continue
```

Comment:
I think that @gagantrivedi could better answer this, since he wrote the be side of metadata, and I made some adjustments.

Reply:
The other implementations of managing metadata for an object use a writeable nested serializer though. Changing this might be significant effort here because of the way in which the segment endpoints work, but I'd certainly prefer to look into it. 

Perhaps for now, to get these PRs over the line, we should just ignore metadata for segments?

Reply:
I think that the segments metadata should not be most used so we could release the feature without metadata for now

Reply:
I appreciate the heads-up and the example. I'll review the test at the provided URL for similar adjustments. Thanks for your input.

Reply:
Ok, I reviewed it, you were right, there was an issue with the SegmentSerializer, now fixed.